### PR TITLE
20170829 UI update

### PR DESCRIPTION
**Pull request by hampgoodwin**
Georg,

Please see the updated UI. 
Thing should be mostly mobile-first ready as well. The CSS is using a twitter bootstrap 4.0.0.1 Alpha framework with a "United" theme. We can change themes around if we don't like the colors.

Noteworthy changes:
- Improved UI look.
- Moved location of CSS and JS scripts within .html files for faster loading, easier future editting.
- Standardized flow and organization of html code.

Pending changes:
- Include the "notice" alert on the stats.html page as well? Should be easy to modify .js and include there as well if we wished.
- bootstrap.js shows an error requiring tether.io as a bootstrap.js depency, however this doesn't affect functionality. We can dig deeper into this issue if we like.

Suggested changes / Observations:
- We discussed briefly if we wanted to use some kind of front-end to handle the display of the scraped data. I think earlier implementation would solve growing-pains if the tools get's larger. 

  - However, doing an iteration of making the "expanding detail" functionality in straight JS may be a useful learning experience for me before transitioning to a front-end framework (Angular, React, Vue).

Let me know what should be changed before accepting the PR.

Also, let me know if there's any kind of "standard" procedure or "best practices" I should include when doing any part of the OS process, I'm especially thinking about Commits, PR's, and documentation.

### CODE REVIEW

**Comment by georgf** [comment]
This looks so much more modern, thanks.
It's great that the notice can be dismissed, that's a nice little detail.
The clear write-up here of the changes and status makes it much easier for me to follow along.

For the framework i filed #5. I think it makes sense to do that first before diving into the detail view.

I took a quick pass on the look and functionality with the changes here applied:
- A theme with more muted colors would be great. We haven't settled on a standard, but usually use more blue/black/gray/white as core colors. E.g. [1](https://telemetry.mozilla.org/new-pipeline/dist.html), [2](https://data-missioncontrol.dev.mozaws.net/), [3](https://analysis.telemetry.mozilla.org/). What do you think of the cosmo theme?
- We should show the notice on the stats page too, to avoid misunderstandings. The "last update" date is not showing on `stats.html` for me.

I'll take a look over the code changes later.

**Comment by hampgoodwin** [comment]
> A theme with more muted colors would be great. We haven't settled on a standard, but usually use more blue/black/gray/white as core colors. E.g. 1, 2, 3. What do you think of the cosmo theme?

Cosmo theme seems great, it looks like what has been applied elsewhere so it would be good to be consistent across similar tools.

> We should show the notice on the stats page too, to avoid misunderstandings. The "last update" date is not showing on stats.html for me.

I moved the global variable #last_updated from explore.js to stats.js so the last_update would be available for both notice alerts. 

I realize there is an issue for combining .js files. However, depending on how we handle the front-end we may not need either? I would be baffled if the front-end frameworks don't **all** have a way to consume/present JSON data.

To push this changes to do, do I PR again, or how does that work?

Also, were you able to determine which front-end we wanted to use? I have a "basic" exposure to the 3 major front-ends (Angular4, React, and Vue) and am anxious to explore any of the three further.

**Comment by georgf** [comment]
> Cosmo theme seems great, it looks like what has been applied elsewhere so it would be good to be consistent across similar tools.

Great, let's use that!
I'll take a pass over the code changes now.

We should keep this PR specific to the bootstrap style changes.
You can use a separate PR that addresses #5 - using a new front-end framework. I added an update on that issues comments.

File: data/index.html
Comment (georgf) [inline data/index.html]: Should this be Cosmo?

File: data/index.html
Comment (georgf) [inline data/index.html]: Nitpicking here: Add a space after the ":", e.g. "JS: jQuery ...".
Ditto below and in stats.html.

File: data/index.html
Comment (georgf) [inline data/index.html]: Instead of commenting out, lets just remove this if it doesn't fit the style.
We can open a follow-up issue to figure out where to link to the GH repo.

File: data/index.html
Comment (georgf) [inline data/index.html:46]: Interesting, lets file a new follow-up issue for that?

File: data/index.html
Comment (georgf) [inline data/index.html]: In `stats.js`, we pass this element to D3, which then draws the visualization into it.
I.e.: `var svg = d3.select("#stats");`

**Review by georgf — CHANGES_REQUESTED** [review]
Thanks, this looks like a big improvement!

In reviewing, i found the atomic commits in this PR pretty helpful to follow along what you are doing.
I have some minor comments here, after fixing that we can merge this PR.

I'm missing front-end know-how, so i can't review the changes in depth - but we can get more experienced review/feedback on that when we move this into an official repository soon.

**Comment by georgf** [comment]
To line out next steps from here:
- Lets settle this PR and merge it.
- We can modernize the front-end per #5.
- (Optionally improve the layout of the search results?)
- Move this into an official repository, we can get front-end review from someone else at this point.
- (Possibly more improvements like #2 & #3)

File: data/index.html
Comment (hampgoodwin) [reply data/index.html]: Changed.

File: data/index.html
Comment (hampgoodwin) [reply data/index.html]: Right. However, D3 is not pushing anything to the element on the index.html page, correct?

File: data/index.html
Comment (hampgoodwin) [reply data/index.html:46]: Yes, let's do that. I think that's something we can address as part of the new front-rend implementation using React+Redux.

File: data/index.html
Comment (hampgoodwin) [reply data/index.html]: Removed.

File: data/index.html
Comment (hampgoodwin) [reply data/index.html]: Updated file name and link reference for clarity.

**Comment by hampgoodwin** [comment]
Updates for requested changes on Aug 30 are implemented.

File: data/index.html
Comment (georgf) [reply data/index.html:46]: I opened #6 for that.

File: data/index.html
Comment (georgf) [reply data/index.html]: Oh, i see!
Right, this seems unused.

File: data/index.html
Comment (georgf) [inline data/index.html]: The closing bracket here got lost.

File: data/index.html
Comment (georgf) [inline data/index.html]: I opened #7 to add this somewhere else later.

**Review by georgf — COMMENTED** [review]
This looks good to me overall, thanks!

One thing that stands out:
On index.html, i don't see "nightly" in the "channel" dropdown, but the [old version](http://georgf.github.io/fx-data-explorer/) has it.

This seems to be just a small markup issue below.

File: data/index.html
Comment (hampgoodwin) [reply data/index.html]: Ah.
Fixed, good catch.

**Review by georgf — APPROVED** [review]
Thanks Abel, this looks good!

**Comment by georgf** [comment]
Next good issues from here could be #5 or #10.

**Pull request merged**
